students/mayc4t/hw3/slicing_lab.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+# #############################################################################
+# Written by: Mayc4t
+# March 19, 2018
+
+# #############################################################################
+
+
+def exchange_first_last(seq):
+    """ Exchange the first and last item. """
+    if(type(seq) is str):
+        stra = seq[-1]
+        for i in range(1, len(seq)-1):
+            stra += seq[i]
+        stra += seq[0]
+        return stra
+    else:
+        new_seq = []
+        new_seq.append(seq[-1])
+        for i in range(1, len(seq)-1):
+            new_seq.append(seq[i])
+        new_seq.append(seq[0])
+        if type(seq) is tuple:
+            new_seq = tuple(new_seq)
+        return new_seq
+
+# #############################################################################
+
+
+def remove_every_other(seq):
+    """ Remove every other item """
+    if(type(seq) is str):
+        new_seq = list(seq)
+        del(new_seq[1:len(new_seq):2])
+        stra = ''
+        for i in range(len(new_seq)):
+            stra += new_seq[i]
+        return stra
+    else:
+        new_seq = list(seq)
+        del(new_seq[1:len(new_seq):2])
+        if type(seq) is tuple:
+            new_seq = tuple(new_seq)
+        return(new_seq)
+
+
+# #############################################################################
+def remove_f4_l4_every_other(seq):
+    """ Remove the first 4 and the last 4 items and then every other item in between."""
+    # check if length of seg has to be at least 10
+    if len(seq) <= 9:
+        print("the seq has to be at last 9 items")
+        pass
+    else:
+        new_seq = list(seq)
+        del(new_seq[0:4:1])
+        del(new_seq[-1:-5:-1])
+        del(new_seq[1: len(new_seq): 2])
+
+        if(type(seq) is str):
+            stra = ''
+            for i in range(len(new_seq)):
+                stra += new_seq[i]
+            return stra
+        else:
+            if type(seq) is tuple:
+                new_seq = tuple(new_seq)
+
+        return(new_seq)
```

Comment:
above functions are not using slicing (`[start:stop:end]`)

all of the 3 above functions are one liners using slicing and will work for any type (tuple, list or string)